spyql/cli.py

```diff
@@ -207,6 +207,42 @@ def parse_select(sel, strings):
     return res, has_distinct, has_partials
 
 
+def parse_wherelike(clause, strings):
+    """splits the LIKE clause and completely supports the SQL syntax
+    https://docs.microsoft.com/en-us/sql/t-sql/language-elements/like-transact-sql?view=sql-server-ver15"""
+    # We're not in a LIKE expression, do nothing
+    if not re.search("LIKE", clause):
+        return clause
+
+    # Supports words containing [a-zA-Z0-9_\-]
+    expr_pattern = re.compile(r"([\w-]+)(?:\s+(NOT))?\s+LIKE\s+([\w-]+)", re.IGNORECASE)
+    groups = re.search(expr_pattern, clause)
+    if groups is None:
+        spyql.log.user_error(
+            f"{clause}",
+            SyntaxError("unexpected EOF while parsing")
+        )
+
+    groups = groups.groups()
+    negate = "NOT" in {groups[1]} # placed within {} because it can be None
+
+    if not groups[2] in strings:
+        spyql.log.user_error(
+            f"{groups[2]}: missing quotes, must be a string",
+            SyntaxError("bad query")
+        )
+
+    # Replacing SQL wildcard '%' for regex wildcard '.*' if not preceded by '\'
+    pattern = strings.put_strings_back(groups[2])
+    pattern = re.compile(r"(?<!\\)%").sub(r".*" , pattern)
```

Comment:
shouldn't we first escape any regex special character/command? For instance `col1 like '123.456.%'` would accept '1233456oops' because of the meaning of `.` in regex patterns...